subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -538,7 +540,7 @@ public String getName() {
 
         @Override
         public boolean isPresent() {
-            return findByNameWithoutRules(name) != null;
+            return findTask(name) != null;
```

Comment:
❌ I think this should consider the following:
- If a provider was removed, this return `false` (but if the provider was replaced by an eager task, this should probably return `true` pointing to the new task... maybe... actually yes because of this: https://github.com/gradle/gradle/pull/5496/files#diff-c1f28c1e4f6ce6dd164dd64d233b8f2fR583)
- If the resolution generated an exception at creation, this return `false`
- If the resolution generated an exception during configuration, this return `true` (because the task is not present in the container)